modules/poem.py

```diff
@@ -99,6 +99,42 @@ def iosock_try():
 
     return True
 
+def check_metric_conf(profiles, arguments):
+
+    # Load metrics from api...
+    try:
+        metrics = requests.get('https://' + arguments.hostname + '//poem/api/0.2/json/metrics/?tag=production',
+                               cert=(arguments.cert, arguments.key), verify=True)
+        metrics = metrics.json()
+    except requests.exceptions.RequestException as e:
+        print 'CRITICAL - cannot connect to %s: %s' % ('https://' + arguments.hostname +
+                                                       '//poem/api/0.2/json/metrics/?tag=production',
+                                                       errmsg_from_excp(e))
+        raise SystemExit(2)
+    except ValueError as e:
+        print 'CRITICAL - %s - %s' % (MIP_API, errmsg_from_excp(e))
+        raise SystemExit(2)
+
+    # Extract metrics in a certain profile...
+    profile_metrics = set()
+    for i in profiles:
+        if arguments.profile in i['description']:
+            for j in i['metric_instances']:
+                profile_metrics.add(j['metric'])
+
+    # Extract metric names...
+    metrics_name = set()
+    for k in metrics:
+        for key, value in k.items():
+            metrics_name.add(key)
+
+    # Check configurations exist...
+    diff = profile_metrics.difference(metrics_name)
```

Comment:
Refactor this to lookup for all elements from metrics_name set existing in profile_metrics set.